### PR TITLE
Initial Commit

### PR DESCRIPTION
**Pull request by cs-shubham-morankar**
#### Actions:
- Run Search

#### UTCs: 

- [x] Connector installation verified.
- [x] Connector image verified.
- [x] Actions and Playbooks list verified.
- [x] Check health verified (Basic Sanity).
- [x] Playbook tags verified. 
- [x] Run Search action verified (Basic Sanity).

### CODE REVIEW

File: bmc-discovery/info.json [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+          "editable": true,
+          "visible": true,
+          "type": "checkbox",
+          "name": "delete",
+          "tooltip": "(Optional) Controls whether the result set is deleted or not before results are returned.\nResult sets are deleted automatically before the last page of results are returned, but setting this field to true on any page will delete the result set immediately.\nThis is an optimization to free resources if you don't plan on accessing any further pages of data. Alternatively, if set to false, automatic deletion of the result set will not happen (for this request).",
+          "description": "(Optional) Controls whether the result set is deleted or not before results are returned.\nResult sets are deleted automatically before the last page of results are returned, but setting this field to true on any page will delete the result set immediately.\nThis is an optimization to free resources if you don't plan on accessing any further pages of data. Alternatively, if set to false, automatic deletion of the result set will not happen (for this request).",
+          "value": false
+        }
+      ],
+      "output_schema": {}
```
Comment (cs-utkarsh-surana) [inline bmc-discovery/info.json:118]: Output schema is missing.

File: bmc-discovery/info.json [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      }
+    ]
+  },
+  "operations": [
+    {
+      "title": "Run Search",
+      "operation": "search_query",
+      "category": "investigation",
+      "annotation": "search_query",
+      "description": "",
```
Comment (cs-utkarsh-surana) [inline bmc-discovery/info.json]: Description missing

File: bmc-discovery/bmc_api_auth.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  FORTINET CONFIDENTIAL & FORTINET PROPRIETARY SOURCE CODE
+  Copyright end """
+
+import requests
+from time import time, ctime
+from datetime import datetime
+from connectors.core.connector import get_logger, ConnectorError
+from connectors.core.utils import update_connnector_config
+
+CONFIG_SUPPORTS_TOKEN = True
```
Comment (cs-utkarsh-surana) [inline bmc-discovery/bmc_api_auth.py]: No use

File: bmc-discovery/bmc_api_auth.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+            error_msg = response.json()
+            raise ConnectorError(error_msg)
+    except Exception as err:
+        raise ConnectorError(err)
+
+
+def check(config, connector_info):
+    try:
+        bmc = BMCAuth(config)
+        if CONFIG_SUPPORTS_TOKEN:
```
Comment (cs-utkarsh-surana) [inline bmc-discovery/bmc_api_auth.py]: No need

File: bmc-discovery/operations.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        token = bmc.validate_token(config, connector_info)
+        return token
+    except Exception as err:
+        raise ConnectorError(err)
+
+
+def make_api_call(config, url, method='GET', params=None, body=None, connector_info=None):
+    try:
+        endpoint = config.get('url') + '/api/v1.1' + url
+        token = get_token(config, connector_info)
```
Comment (cs-utkarsh-surana) [inline bmc-discovery/operations.py]: validate token can be directly called here.

**Review by cs-utkarsh-surana — COMMENTED** [review]
Please make necessary changes.

File: bmc-discovery/info.json [hunk 6]
```diff
@@ -0,0 +1,121 @@
+{
+  "name": "bmc-discovery",
+  "label": "BMC Discovery",
+  "description": "BMC Discovery is a data center discovery solution that automatically discovers data center inventory, configuration and relationship data, and maps applications to the IT infrastructure. This connector facilitates automated operation related to",
```
Comment (cs-prerna-joshi) [inline bmc-discovery/info.json]: incomplete description.

File: bmc-discovery/info.json [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+  "publisher": "Community",
```
Comment (cs-prerna-joshi) [inline bmc-discovery/info.json]:
```suggestion
  "publisher": "Fortinet",
```

File: bmc-discovery/info.json [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+        {
+          "title": "Filter Query",
+          "required": true,
+          "editable": true,
+          "visible": true,
+          "type": "text",
+          "name": "query",
+          "tooltip": "Specify a search query to run, in BMC Discovery query language.",
+          "description": "Specify a search query to run, in BMC Discovery query language.",
+          "placeholder": "For e.g. SEARCH Host WHERE os_type = 'Windows' ORDER BY name SHOW name, ram"
```
Comment (cs-prerna-joshi) [inline bmc-discovery/info.json]:
```suggestion
          "placeholder": "e.g. SEARCH Host WHERE os_type = 'Windows' ORDER BY name SHOW name, ram"
```

File: bmc-discovery/info.json (hunk 2 above)
Comment (cs-shubham-morankar) [reply bmc-discovery/info.json]: We don't have proper response in API doc.

File: bmc-discovery/info.json (hunk 1 above)
Comment (cs-shubham-morankar) [reply bmc-discovery/info.json:118]: We don't have proper response in API doc that's why we keep as empty.

**Review by cs-prerna-joshi — APPROVED** [review]

**Pull request merged**